# -*- coding: utf-8 -*-
"""Zad.6

Automatically generated by Colaboratory.

Original file is located at
    https://colab.research.google.com/drive/1Xc_0fMvExW9RG5uC34S7XPH3ULvHBXWg
"""

def process_lists_and_return_result(list1, list2):
    if not all(isinstance(lst, list) for lst in [list1, list2]):
        raise TypeError("Both arguments must be of type list")

    merged_list = list(set(list1 + list2))
    processed_list = [x**3 for x in merged_list]

    return processed_list

try:
    list1 = [12, 2, 3, 4]
    list2 = [3, 4, 5, 6]

    result = process_lists_and_return_result(list1, list2)

    print("Wynik po połączeniu list, usunięciu duplikatów i podniesieniu do potęgi 3:")
    print(result)
except TypeError as e:
    print(f"Błąd: {e}")